app/data/release-schedule.ts

```diff
@@ -0,0 +1,231 @@
+import { parse as parseSemver } from 'semver';
+import memoize from '@keyvhq/memoize';
+import { ElectronRelease, getReleasesOrUpdate } from './release-data';
+import { extractChromiumMilestone, getPrereleaseType } from '~/helpers/version';
+import { getMilestoneSchedule } from './dash/chromium-schedule';
+import { getKeyvCache } from './cache';
+
+export interface MajorReleaseSchedule {
+  version: string; // `${major}.0.0`
+  alphaDate: string | null; // YYYY-MM-DD -- some old versions didn't have alpha releases
+  betaDate: string; // YYYY-MM-DD
+  stableDate: string; // YYYY-MM-DD
+  eolDate: string; // YYYY-MM-DD
+  chromiumVersion: number; // milestone, aka major version
+  nodeVersion: string; // full semver
+  status: 'active' | 'prerelease' | 'eol';
+}
+
+type AbsoluteMajorReleaseSchedule = Omit<MajorReleaseSchedule, 'status'>;
+
+interface MajorReleaseGroup {
+  major: number;
+  releases: ElectronRelease[];
+  firstStable?: ElectronRelease; // Only used for Chromium milestone extraction
+}
+
+const SCHEDULE_OVERRIDES: Map<string, Partial<AbsoluteMajorReleaseSchedule>> = new Map([
```

Comment:
I think we need more adjustments here, there's more discrepancies with the dates at https://www.electronjs.org/docs/latest/tutorial/electron-timelines for the historical dates, particularly betas.

When I was attempting this I was [pulling those historical dates from the releases data](https://github.com/electron/release-status/compare/44c966501cc724d2909caec55ead9ef4bbde47e5...versions-json-refactor#diff-8190b0aaf866b637d1461f25e6c188f27afce6da67cac9f2bb9362790ce02110R92-R103). Maybe you want to consider doing that? The actual release dates in that file should take precedence over the calculated schedule dates, and would require fewer overrides.